Automattic/wp-calypso
# Plans 2023: Migrate PlanPill under plans-grid component

Fixes https://github.com/Automattic/wp-calypso/issues/79009

## Proposed Changes

* Makes `PlanPill` a local component in `plans-grid` package
* Updates references and CSS class name to match
* Updates dependencies
* No other cleanup

### Media

<img width="400" alt="Screenshot 2023-10-26 at 3 18 54 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/57c7eedb-0c59-4e0f-aa87-1bcb1441da17">

## Testing Instructions



* Go to `/plans` and `/start/plans`
* Confirm popular badges render as expected

## Pre-merge Checklist
